Add dynamo_test_failures.csv; change update_failures.py to update this

zou3519 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #119572

This PR:
- moves the test failure lists from dynamo_test_failures.py to
  dynamo_test_failures.csv
- Every line in dynamo_test_failures.csv is separated by 3 empty lines
  and all the lines are sorted.
- changes update_failures.py to update dynamo_test_failures.csv

The motivation for this change is avoiding merge conflicts when multiple
people touch dynamo_test_failures.csv. git is good at resolving the
conflicts when there is sufficient empty space around.

Test Plan:
- tested locally (we should probably write some tests for
  update_failures.py...)

cc @albanD

zou3519 committed:
Add dynamo_test_failures.csv; change update_failures.py to update this

This PR:
- moves the test failure lists from dynamo_test_failures.py to
  dynamo_test_failures.csv
- Every line in dynamo_test_failures.csv is separated by 3 empty lines
  and all the lines are sorted.
- changes update_failures.py to update dynamo_test_failures.csv

The motivation for this change is avoiding merge conflicts when multiple
people touch dynamo_test_failures.csv. git is good at resolving the
conflicts when there is sufficient empty space around.

Test Plan:
- tested locally (we should probably write some tests for
  update_failures.py...)

[ghstack-poisoned]

zou3519 committed:
Update on "Add dynamo_test_failures.csv; change update_failures.py to update this"


This PR:
- moves the test failure lists from dynamo_test_failures.py to
  dynamo_test_failures.csv
- Every line in dynamo_test_failures.csv is separated by 3 empty lines
  and all the lines are sorted.
- changes update_failures.py to update dynamo_test_failures.csv

The motivation for this change is avoiding merge conflicts when multiple
people touch dynamo_test_failures.csv. git is good at resolving the
conflicts when there is sufficient empty space around.

Test Plan:
- tested locally (we should probably write some tests for
  update_failures.py...)

cc albanD

[ghstack-poisoned]

zou3519 committed:
Update on "Add dynamo_test_failures.csv; change update_failures.py to update this"


This PR:
- moves the test failure lists from dynamo_test_failures.py to
  dynamo_test_failures.csv
- Every line in dynamo_test_failures.csv is separated by 3 empty lines
  and all the lines are sorted.
- changes update_failures.py to update dynamo_test_failures.csv

The motivation for this change is avoiding merge conflicts when multiple
people touch dynamo_test_failures.csv. git is good at resolving the
conflicts when there is sufficient empty space around.

Test Plan:
- tested locally (we should probably write some tests for
  update_failures.py...)

cc albanD

[ghstack-poisoned]

albanD commented on torch/testing/_internal/dynamo_test_failures.csv:
```diff
@@ -0,0 +1,12461 @@
+
+
+
+ActivationCheckpointingTests.test_cond_with_kwargs,xfail,dynamo/test_higher_order_ops
```
Can we avoid the 3 empty lines between each line to reduce the line count by 4x?

oulgen replied:
Without the empty lines, wouldn't we still run into merge conflict?

aorenste replied:
I also would prefer not to have the empty lines. I suspect that simply sorting the lines will eliminate most of the conflicts without all the empty lines. If we still have conflicts after that now that it's a separate file it would be easy enough to write a custom merge driver for this.

zou3519 replied:
The empty lines eliminate merge conflicts without needing a custom merge driver. We've [done this in other parts of PyTorch](https://github.com/pytorch/pytorch/tree/main/benchmarks/dynamo/ci_expected_accuracy) (but not with this many lines) What's the concern around the additional empty lines?

aorenste replied:
It's not a huge deal but the empty lines makes it harder to look at the data visually, makes it harder to sort/edit manually without a custom sorting tool. If we're doing this same thing in many places maybe that's a sign that we should just write a common merge driver...

zou3519 committed:
Update on "Add dynamo_test_failures.csv; change update_failures.py to update this"


This PR:
- moves the test failure lists from dynamo_test_failures.py to
  dynamo_test_failures.csv
- Every line in dynamo_test_failures.csv is separated by 3 empty lines
  and all the lines are sorted.
- changes update_failures.py to update dynamo_test_failures.csv

The motivation for this change is avoiding merge conflicts when multiple
people touch dynamo_test_failures.csv. git is good at resolving the
conflicts when there is sufficient empty space around.

Test Plan:
- tested locally (we should probably write some tests for
  update_failures.py...)

cc albanD

[ghstack-poisoned]

zou3519 committed:
Update on "Add dynamo_test_failures.csv; change update_failures.py to update this"


This PR:
- moves the test failure lists from dynamo_test_failures.py to
  dynamo_test_failures.csv
- Every line in dynamo_test_failures.csv is separated by 3 empty lines
  and all the lines are sorted.
- changes update_failures.py to update dynamo_test_failures.csv

The motivation for this change is avoiding merge conflicts when multiple
people touch dynamo_test_failures.csv. git is good at resolving the
conflicts when there is sufficient empty space around.

Test Plan:
- tested locally (we should probably write some tests for
  update_failures.py...)

cc albanD

[ghstack-poisoned]

zou3519 committed:
Update on "Add dynamo_test_failures.csv; change update_failures.py to update this"


This PR:
- moves the test failure lists from dynamo_test_failures.py to
  dynamo_test_failures.csv
- Every line in dynamo_test_failures.csv is separated by 3 empty lines
  and all the lines are sorted.
- changes update_failures.py to update dynamo_test_failures.csv

The motivation for this change is avoiding merge conflicts when multiple
people touch dynamo_test_failures.csv. git is good at resolving the
conflicts when there is sufficient empty space around.

Test Plan:
- tested locally (we should probably write some tests for
  update_failures.py...)

cc albanD

[ghstack-poisoned]

zou3519 committed:
Update on "Add dynamo_test_failures.csv; change update_failures.py to update this"


This PR:
- moves the test failure lists from dynamo_test_failures.py to
  dynamo_test_failures.csv
- Every line in dynamo_test_failures.csv is separated by 3 empty lines
  and all the lines are sorted.
- changes update_failures.py to update dynamo_test_failures.csv

The motivation for this change is avoiding merge conflicts when multiple
people touch dynamo_test_failures.csv. git is good at resolving the
conflicts when there is sufficient empty space around.

Test Plan:
- tested locally (we should probably write some tests for
  update_failures.py...)

cc albanD

[ghstack-poisoned]
